cppsdk/include/playfab/PlayFabEventApi.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#ifndef DISABLE_PLAYFABENTITY_API
+
+#include <playfab/PlayFabEvent.h>
+#include <playfab/PlayFabEventRouter.h>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Main interface for PlayFab Sdk, specifically all Lightweight/Heavyweight Event APIs
+    /// </summary>
+    class PlayFabEventAPI
+    {
+    public:
+        PlayFabEventAPI();
+        void EmitEvent(std::unique_ptr<const IPlayFabEvent> event, const PlayFabEmitEventCallback callback) const;
```

Comment:
Should we indicate if the emission was successful ? Or is this implicit via callback?

Reply:
It am thinking to make it via callback.

---
In reply to: [227948999](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227948999) [](ancestors = 227948999)

Reply:
Are there scenarios where we would have to drop the event on the floor due to space limitations of the pipeline? Insertion into the queue fails so we want to return false?

Reply:
That would be returned in callback but we can discuss that

---
In reply to: [227990565](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227990565) [](ancestors = 227990565)